### ISSUE TITLE
storage: panic in GenerateFlamegraph

### ISSUE THREAD

**Issue by oncilla** [parca-dev/parca#321]
Panic trace run through [panicparsed](https://github.com/maruel/panicparse):
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x113cf6b]


To see all goroutines, visit https://github.com/maruel/panicparse#gotraceback

1: running [Created by grpc.(*Server).serveStreams.func1 @ server.go:921]
    storage            flamegraph.go:121     GenerateFlamegraph(#3, #12, 0x194b220, 0x22c03d8, 0x7f16a7603698, 0xc00000e7e0, 0, 0, 0, 0, ...)
    query              query.go:273          (*Query).renderReport(#6, #3, #12, 0, 0, 0, 0, 0, #3)
    query              query.go:220          (*Query).mergeRequest(#6, #3, #12, 0xc0146be820, 0, 0, 0)
    query              query.go:155          (*Query).Query(#6, #3, #11, #13, #6, 0x13fc240, 0xc0146a4cb0)
    v1alpha1           query_grpc.pb.go:164  _QueryService_Query_Handler.func1(#3, #11, #1, #13, 0, 0x13c4f9, 0x5, 0x166630d)
    interceptors       server.go:22          UnaryServerInterceptor.func1(#3, #11, #1, #13, #14, #9, 0x3, 0x3, 0x1970160, 0xc0026b8de0)
    go-grpc-middleware chain.go:25           ChainUnaryServer.func1.1.1(#3, #11, #1, #13, 0x28, 0xc0146bec30, #2, 0x8)
    go-grpc-prometheus server_metrics.go:107 (*ServerMetrics).UnaryServerInterceptor.func1(#3, #11, #1, #13, #14, 0xc0146e24e0, 0, 0, 0, 0)
    go-grpc-middleware chain.go:25           ChainUnaryServer.func1.1.1(#3, #11, #1, #13, #2, 0x8, 0, 0)
    otelgrpc           interceptor.go:325    UnaryServerInterceptor.func1(#3, #11, #1, #13, #14, 0xc0146e2500, 0, 0, 0, 0)
    go-grpc-middleware chain.go:25           ChainUnaryServer.func1.1.1(#3, #10, #1, #13, 0xc000182800, 0xc000637980, 0xc013979b30, 0x40e318)
    go-grpc-middleware chain.go:34           ChainUnaryServer.func1(#3, #10, #1, #13, #14, #9, 0x4723a7, 0x4a2a65, 0x155fb80, #10)
    v1alpha1           query_grpc.pb.go:166  _QueryService_Query_Handler(0x15a4500, #6, #3, #10, 0xc0146bc2a0, 0xc0001645a0, #3, #10, 0xc0146f6000, 0x4d)
    grpc               server.go:1279        (*Server).processUnaryRPC(#5, #4, #7, #8, 0xc000164840, 0x227af98, 0, 0, 0)
    grpc               server.go:1608        (*Server).handleStream(#5, #4, #7, #8, 0)
    grpc               server.go:923         (*Server).serveStreams.func1.2(0xc00cda6ba0, #5, #4, #7, #8)
```

Original:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x113cf6b]

goroutine 52228 [running]:
github.com/parca-dev/parca/pkg/storage.GenerateFlamegraph(0x19676d0, 0xc0146bb260, 0x194b220, 0x22c03d8, 0x7f16a7603698, 0xc00000e7e0, 0x0, 0x0, 0x0, 0x0, ...)
        github.com/parca-dev/parca/pkg/storage/flamegraph.go:121 +0x16b
github.com/parca-dev/parca/pkg/query.(*Query).renderReport(0xc0002d2300, 0x19676d0, 0xc0146bb260, 0x0, 0x0, 0x0, 0x0, 0x0, 0x19676d0)
        github.com/parca-dev/parca/pkg/query/query.go:273 +0xda
github.com/parca-dev/parca/pkg/query.(*Query).mergeRequest(0xc0002d2300, 0x19676d0, 0xc0146bb260, 0xc0146be820, 0x0, 0x0, 0x0)
        github.com/parca-dev/parca/pkg/query/query.go:220 +0x1f0
github.com/parca-dev/parca/pkg/query.(*Query).Query(0xc0002d2300, 0x19676d0, 0xc0146bb170, 0xc0146be5f0, 0xc0002d2300, 0x13fc240, 0xc0146a4cb0)
        github.com/parca-dev/parca/pkg/query/query.go:155 +0x185
github.com/parca-dev/parca/gen/proto/go/parca/query/v1alpha1._QueryService_Query_Handler.func1(0x19676d0, 0xc0146bb170, 0x158bb60, 0xc0146be5f0, 0x0, 0x13c4f9, 0x5, 0x166630d)
        github.com/parca-dev/parca/gen/proto/go/parca/query/v1alpha1/query_grpc.pb.go:164 +0x89
github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors.UnaryServerInterceptor.func1(0x19676d0, 0xc0146bb170, 0x158bb60, 0xc0146be5f0, 0xc0146e24c0, 0xc0146b6420, 0x3, 0x3, 0x1970160, 0xc0026b8de0)
        github.com/grpc-ecosystem/go-grpc-middleware/v2@v2.0.0-rc.2.0.20201207153454-9f6bf00c00a7/interceptors/server.go:22 +0x239
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1(0x19676d0, 0xc0146bb170, 0x158bb60, 0xc0146be5f0, 0x28, 0xc0146bec30, 0x1632679, 0x8)
        github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x63
github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1(0x19676d0, 0xc0146bb170, 0x158bb60, 0xc0146be5f0, 0xc0146e24c0, 0xc0146e24e0, 0x0, 0x0, 0x0, 0x0)
        github.com/grpc-ecosystem/go-grpc-prometheus@v1.2.0/server_metrics.go:107 +0xb0
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1(0x19676d0, 0xc0146bb170, 0x158bb60, 0xc0146be5f0, 0x1632679, 0x8, 0x0, 0x0)
        github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x63
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.UnaryServerInterceptor.func1(0x19676d0, 0xc0146bb170, 0x158bb60, 0xc0146be5f0, 0xc0146e24c0, 0xc0146e2500, 0x0, 0x0, 0x0, 0x0)
        go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.24.0/interceptor.go:325 +0x6b6
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1(0x19676d0, 0xc0146ba960, 0x158bb60, 0xc0146be5f0, 0xc000182800, 0xc000637980, 0xc013979b30, 0x40e318)
        github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x63
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1(0x19676d0, 0xc0146ba960, 0x158bb60, 0xc0146be5f0, 0xc0146e24c0, 0xc0146b6420, 0x4723a7, 0x4a2a65, 0x155fb80, 0xc0146ba960)
        github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34 +0xd7
github.com/parca-dev/parca/gen/proto/go/parca/query/v1alpha1._QueryService_Query_Handler(0x15a4500, 0xc0002d2300, 0x19676d0, 0xc0146ba960, 0xc0146bc2a0, 0xc0001645a0, 0x19676d0, 0xc0146ba960, 0xc0146f6000, 0x4d)
        github.com/parca-dev/parca/gen/proto/go/parca/query/v1alpha1/query_grpc.pb.go:166 +0x150
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000207c00, 0x1975b98, 0xc010a38ab0, 0xc0143a0900, 0xc000164840, 0x227af98, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.41.0/server.go:1279 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000207c00, 0x1975b98, 0xc010a38ab0, 0xc0143a0900, 0x0)
        google.golang.org/grpc@v1.41.0/server.go:1608 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00cda6ba0, 0xc000207c00, 0x1975b98, 0xc010a38ab0, 0xc0143a0900)
        google.golang.org/grpc@v1.41.0/server.go:923 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.41.0/server.go:921 +0x1fd

**Comment by thorfour** [comment]
This panic should be resolved with #330

**Comment by oncilla** [comment]
Thanks for the swift fix.

I don't remember exactly what I did, but I think it was something like this:

I was running it in a testing environment where it is part of a docker-compose topology. I think I kept the browser open and destroyed the topo and started a new one, and then reloaded the page (or pressed a button?).
Which would kind of match with what was fixed in #330, as the the data from the previous run was not available anymore.
